unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ChatWidgetHUD/ChatEntry.cs

```diff
@@ -133,42 +133,42 @@ public void Populate(Model chatEntryModel)
         if (fadeEnabled)
             group.alpha = 0;
 
-        if (HUDAudioPlayer.i != null)
+        if (HUDAudioHandler.i != null)
         {
             // Check whether or not this message is new
-            if (chatEntryModel.timestamp > HUDAudioPlayer.i.chatLastCheckedTimestamp)
+            if (chatEntryModel.timestamp > HUDAudioHandler.i.chatLastCheckedTimestamp)
             {
                 switch (chatEntryModel.messageType)
                 {
                     case ChatMessage.Type.PUBLIC:
                         // Check whether or not the message was sent by the local player
                         if (chatEntryModel.senderId == UserProfile.GetOwnUserProfile().userId)
-                            HUDAudioPlayer.i.Play(HUDAudioPlayer.Sound.sendChatEntry);
+                            AudioScriptableObjects.chatSend.Play(true);
                         else
-                            HUDAudioPlayer.i.Play(HUDAudioPlayer.Sound.receiveGlobalChatEntry);
+                            AudioScriptableObjects.chatReceiveGlobal.Play(true);
                         break;
                     case ChatMessage.Type.PRIVATE:
                         switch (chatEntryModel.subType)
                         {
                             case Model.SubType.PRIVATE_FROM:
-                                HUDAudioPlayer.i.Play(HUDAudioPlayer.Sound.receivePrivateChatEntry);
+                                AudioScriptableObjects.chatReceivePrivate.Play(true);
                                 break;
                             case Model.SubType.PRIVATE_TO:
-                                HUDAudioPlayer.i.Play(HUDAudioPlayer.Sound.sendChatEntry);
+                                AudioScriptableObjects.chatSend.Play(true);
                                 break;
                             default:
                                 break;
                         }
                         break;
                     case ChatMessage.Type.SYSTEM:
-                        HUDAudioPlayer.i.Play(HUDAudioPlayer.Sound.receiveGlobalChatEntry);
+                        AudioScriptableObjects.chatReceiveGlobal.Play(true);
                         break;
                     default:
                         break;
                 }
             }
 
-            HUDAudioPlayer.i.RefreshChatLastCheckedTimestamp();
+            HUDAudioHandler.i.RefreshChatLastCheckedTimestamp();
```

Comment:
I think we can move this `RefreshChatLastCheckedTimestamp` to this class, and remove `HUDAudioHandler` completely. 😄

Reply:
Yes! I'm a bit unclear on which of these I should use. I'll experiment and see if I can figure it out.
![image](https://user-images.githubusercontent.com/1711394/92610239-c4a05400-f2b7-11ea-897b-02f2aa660851.png)

Reply:
I think you can use the `lastReadWorldChatMessages` one.